woocommerce/woocommerce-ios
# System Status Report: Update Networking to separate SystemPluginMapper from SystemStatusMapper

Part of #5071 


### Description

Currently `SystemStatusMapper` is decoding response from `GET /wp-json/wc/v3/system_status` and returns a plugin list from there since previous usage includes only information about plugins.

For #5071 we want to get all information about system status, so this PR separates the old mapper into 2 different ones: one for plugins only and one for all information.

This PR also adds new type `Environment` for system status and updates unit tests and JSON file, hence the large number of file changes. Other changes will be added in another PR to avoid bloating this PR even more!

### Testing instructions

- Make sure that CI (all unit tests) pass
- If possible, please follow the testing steps in #5341 to make sure that system plugins are still loaded correctly.

### Screenshots

N/A

---
